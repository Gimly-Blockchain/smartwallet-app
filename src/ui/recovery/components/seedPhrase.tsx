import React from 'react'
import { StyleSheet, View, Text } from 'react-native'
import { Container } from 'src/ui/structure'
import I18n from 'src/locales/i18n'
import strings from '../../../locales/strings'
import { Colors, Typography, Buttons, Spacing } from 'src/styles'

const { Button } = require('react-native-material-ui')

interface Props {
  seedPhrase: string
  handleButtonTap: () => void
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.blackMain,
  },
  noteSection: {
    marginTop: Spacing.LG,
    flex: 0.8,
    justifyContent: 'center',
  },
  note: {
    ...Typography.subMainText,
    ...Typography.centeredText,
    lineHeight: Typography.subMainText.fontSize + 2,
    color: Colors.sandLight,
  },
  phraseSection: {
    flex: 1,
  },
  seedPhrase: {
    ...Typography.baseFontStyles,
    ...Typography.centeredText,
    fontSize: Typography.text3XL,
    lineHeight: Typography.text3XL + 4,
    color: Colors.white,
  },
  buttonSection: {
    marginTop: 'auto',
  },
  buttonContainer: {
    ...Buttons.buttonStandardContainer,
  },
  buttonText: {
    ...Buttons.buttonStandardText,
  },
})

export const SeedPhrase: React.SFC<Props> = ({
  seedPhrase,
  handleButtonTap,
}: Props) => (
  <Container style={styles.container}>
    <View style={styles.noteSection}>
      <Text style={styles.note}>
        {I18n.t(strings.WRITE_THESE_WORDS_DOWN_ON_AN_ANALOG_AND_SECURE_PLACE) +
          '. ' +
          I18n.t(
            strings.WITHOUT_THESE_WORDS_YOU_CANNOT_ACCESS_YOUR_WALLET_AGAIN,
          ) +
          '.'}
      </Text>
    </View>
    <View style={styles.phraseSection}>
      <Text style={styles.seedPhrase}>{seedPhrase}</Text>
    </View>
    <View style={styles.buttonSection}>
      <Button
        style={{ container: styles.buttonContainer, text: styles.buttonText }}
        onPress={handleButtonTap}
        raised
        upperCase={false}
        text={I18n.t(strings.YES_I_WROTE_IT_DOWN)}
      />
    </View>
  </Container>
)
