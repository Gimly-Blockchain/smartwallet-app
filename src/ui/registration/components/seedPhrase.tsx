import React from 'react'
import { StyleSheet, View, Text } from 'react-native'
import { JolocomTheme } from 'src/styles/jolocom-theme'
import { Container } from 'src/ui/structure/'
import I18n from 'src/locales/i18n'
import strings from '../../../locales/strings'

const { Button } = require('react-native-material-ui')

interface Props {
  seedPhrase: string
  checked: boolean
  handleButtonTap: () => void
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: JolocomTheme.primaryColorBlack,
  },
  noteSection: {
    marginTop: 20,
    flex: 0.8,
    justifyContent: 'center',
  },
  note: {
    textAlign: 'center',
    lineHeight: 26,
    color: JolocomTheme.primaryColorSand,
    fontSize: JolocomTheme.labelFontSize,
    fontFamily: JolocomTheme.contentFontFamily,
  },
  phraseSection: {
    flex: 1,
  },
  seedPhrase: {
    textAlign: 'center',
    color: JolocomTheme.primaryColorWhite,
    fontSize: 34,
    fontFamily: JolocomTheme.contentFontFamily,
    lineHeight: 38,
  },
  buttonSection: {
    marginTop: 'auto',
  },
  buttonContainer: {
    borderRadius: 4,
    height: 48,
    paddingHorizontal: 25,
    backgroundColor: JolocomTheme.primaryColorPurple,
  },
  buttonText: {
    paddingVertical: 15,
    fontWeight: '100',
    fontSize: JolocomTheme.headerFontSize,
    color: JolocomTheme.primaryColorWhite,
    fontFamily: JolocomTheme.contentFontFamily,
  },
})

export const SeedPhrase: React.SFC<Props> = props => (
  <Container style={styles.container}>
    <View style={styles.noteSection}>
      <Text style={styles.note}>
        {I18n.t(strings.WRITE_THESE_WORDS_DOWN_ON_AN_ANALOG_AND_SECURE_PLACE) +
          '. ' +
          I18n.t(
            strings.WITHOUT_THESE_WORDS_YOU_CANNOT_ACCESS_YOUR_WALLET_AGAIN,
          ) +
          '.'}
      </Text>
    </View>
    <View style={styles.phraseSection}>
      <Text style={styles.seedPhrase}>{props.seedPhrase}</Text>
    </View>
    <View style={styles.buttonSection}>
      <Button
        style={{ container: styles.buttonContainer, text: styles.buttonText }}
        onPress={props.handleButtonTap}
        raised
        upperCase={false}
        text={I18n.t(strings.YES_I_WROTE_IT_DOWN)}
      />
    </View>
  </Container>
)
