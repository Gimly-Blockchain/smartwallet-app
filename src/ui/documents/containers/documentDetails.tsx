import React from 'react'
import { connect } from 'react-redux'
import { NavigationScreenProps } from 'react-navigation'
import { View, StyleSheet, ScrollView } from 'react-native'
import { ClaimInterface } from 'cred-types-jolocom-core'
import { RootState } from 'src/reducers/'
import I18n from 'src/locales/i18n'
import { DocumentCard } from '../components/documentCard'
import { DocumentDetails as DocumentDetailsComponent } from '../components/documentDetails'
import strings from '../../../locales/strings'
import { JolocomTheme } from 'src/styles/jolocom-theme.ios'

interface Props
  extends ReturnType<typeof mapStateToProps>,
    NavigationScreenProps {}

interface State {}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: JolocomTheme.primaryColorGrey,
  },
  documentCardContainer: {
    paddingVertical: 15,
    justifyContent: 'center',
    alignItems: 'center',
  },
})

export class DocumentDetailsContainer extends React.Component<Props, State> {
  public static navigationOptions = ({
    navigation,
    navigationOptions,
  }: NavigationScreenProps) => ({
    ...navigationOptions,
    // FIXME needs to be changed to this on update to react-navigation
    // headerTitle: navigation.getParam('headerTitle', ''),
    headerTitle: navigation.state.params && navigation.state.params.headerTitle,
  })

  public componentWillMount() {
    const { document } = this.props
    // FIXME use filter from src/lib/filterDecoratedClaims
    const isExpired = document.expires
      ? document.expires.valueOf() < new Date().valueOf()
      : true

    let headerTitle = (document.claimData as ClaimInterface).type
    if (isExpired) headerTitle = `[${I18n.t(strings.EXPIRED)}] ${headerTitle}`
    this.props.navigation.setParams({ headerTitle })
  }

  public render(): JSX.Element {
    const { document } = this.props

    return (
      <View style={styles.container}>
        <View style={styles.documentCardContainer}>
          <DocumentCard document={document} />
        </View>
        <ScrollView>
          <DocumentDetailsComponent document={document} />
        </ScrollView>
      </View>
    )
  }
}

const mapStateToProps = (state: RootState) => ({
  document: state.documents.selectedDocument,
})

export const DocumentDetails = connect(mapStateToProps)(
  DocumentDetailsContainer,
)
