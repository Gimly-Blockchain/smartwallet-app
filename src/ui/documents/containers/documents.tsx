import React from 'react'
import { connect } from 'react-redux'
import { Animated, ScrollView, StyleSheet, Text, View } from 'react-native'

import { openDocumentDetails } from 'src/actions/documents'
import { DecoratedClaims } from 'src/reducers/account'
import { RootState } from 'src/reducers'
import { getDocumentClaims } from 'src/utils/filterDocuments'
import { ThunkDispatch } from 'src/store'

import I18n from 'src/locales/i18n'
import { filters } from 'src/lib/filterDecoratedClaims'

import { DocumentsCarousel } from '../components/documentsCarousel'
import { DocumentsList } from '../components/documentsList'
import { DocumentViewToggle } from '../components/documentViewToggle'
import strings from '../../../locales/strings'
import { Typography, Colors, Spacing } from 'src/styles'

interface Props
  extends ReturnType<typeof mapDispatchToProps>,
    ReturnType<typeof mapStateToProps> {}

interface State {
  activeDocumentIndex: number
  showingValid: boolean
}

/*
const APPBAR_HEIGHT = Platform.select({
  ios: 44,
  android: 56,
  default: 64,
})
*/

const styles = StyleSheet.create({
  mainContainer: {
    flex: 1,
    backgroundColor: Colors.lightGreyLighter,
  },
  topContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    paddingVertical: Spacing.MD,
  },
  emptyDocumentsContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    paddingHorizontal: Spacing.MD,
  },
  emptyDocumentsText: {
    ...Typography.mainText,
    ...Typography.centeredText,
    color: Colors.greyLight,
  },
})

export class DocumentsContainer extends React.Component<Props, State> {
  public ScrollViewRef: ScrollView | null = null

  public state = {
    activeDocumentIndex: 0,
    showingValid: true,
  }

  private scrollToTop(): void {
    if (this.ScrollViewRef) {
      this.ScrollViewRef.scrollTo({ x: 0, y: 0, animated: true })
    }
  }

  private onActiveIndexChange(index: number): void {
    this.setState({ activeDocumentIndex: index })
    this.scrollToTop()
  }

  private handleToggle = () => {
    this.setState((prevState: State) => ({
      showingValid: !prevState.showingValid,
      activeDocumentIndex: 0,
    }))
    this.scrollToTop()
  }

  public render(): JSX.Element {
    const { openDocumentDetails, decoratedCredentials } = this.props
    const documents = getDocumentClaims(decoratedCredentials['Other'])
    const docFilter = this.state.showingValid
      ? filters.filterByValid
      : filters.filterByExpired
    const displayedDocs = docFilter(documents)
    const isEmpty = displayedDocs.length === 0
    const otherIsEmpty = displayedDocs.length === documents.length

    return (
      <Animated.View style={styles.mainContainer}>
        {!otherIsEmpty && (
          <DocumentViewToggle
            showingValid={this.state.showingValid}
            onTouch={this.handleToggle}
          />
        )}
        <ScrollView
          // scrollEventThrottle={16}
          // onScroll={Animated.event([
          //   {
          //     nativeEvent: {
          //       contentOffset: { y: this.state.headerHeight },
          //     },
          //   },
          // ])}
          // to scroll to top upon changing card
          ref={ref => (this.ScrollViewRef = ref)}
          contentContainerStyle={isEmpty && { flex: 1 }}
          scrollEnabled={!isEmpty}
        >
          {isEmpty ? (
            <View style={styles.emptyDocumentsContainer}>
              <Text style={styles.emptyDocumentsText}>
                {I18n.t(strings.NO_DOCUMENTS_TO_SEE_HERE) + '...'}
              </Text>
            </View>
          ) : this.state.showingValid ? (
            <DocumentsCarousel
              documents={displayedDocs}
              activeIndex={this.state.activeDocumentIndex}
              onActiveIndexChange={this.onActiveIndexChange.bind(this)}
            />
          ) : (
            <DocumentsList
              documents={displayedDocs}
              onDocumentPress={openDocumentDetails}
            />
          )}
        </ScrollView>
      </Animated.View>
    )
  }
}

const mapStateToProps = ({
  account: {
    claims: { decoratedCredentials },
  },
}: RootState) => ({
  decoratedCredentials,
})

const mapDispatchToProps = (dispatch: ThunkDispatch) => ({
  openDocumentDetails: (doc: DecoratedClaims) =>
    dispatch(openDocumentDetails(doc)),
})

export const Documents = connect(
  mapStateToProps,
  mapDispatchToProps,
)(DocumentsContainer)
