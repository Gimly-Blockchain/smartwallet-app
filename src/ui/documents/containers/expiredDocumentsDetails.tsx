import React from 'react'
import { ExpiredDocumentsDetailsComponent } from 'src/ui/documents/components/expiredDocumentDetails'
import { connect } from 'react-redux'
import { RootState } from 'src/reducers/'
import { NavigationScreenProps } from 'react-navigation'
import { DecoratedClaims } from 'src/reducers/account'

interface ConnectProps {
  selectedExpiredDocument: DecoratedClaims
}

interface Props extends ConnectProps {
  documents: Document[]
}

interface State {}

export class ExpiredDocumentsDetailsContainer extends React.Component<
  Props,
  State
> {
  public static navigationOptions = ({
    navigation,
    navigationOptions,
  }: NavigationScreenProps) => ({
    ...navigationOptions,
    headerTitle: navigation.getParam('headerTitle', 'DEFAULT'),
  })

  public render(): JSX.Element {
    const { selectedExpiredDocument } = this.props
    return (
      <ExpiredDocumentsDetailsComponent document={selectedExpiredDocument} />
    )
  }
}

const mapStateToProps = (state: RootState): {} => ({
  selectedExpiredDocument: state.documents.selectedExpiredDocument,
})

const mapDispatchToProps = (dispatch: Function): {} => ({})

export const ExpiredDocumentsDetails = connect(
  mapStateToProps,
  mapDispatchToProps,
)(ExpiredDocumentsDetailsContainer)
