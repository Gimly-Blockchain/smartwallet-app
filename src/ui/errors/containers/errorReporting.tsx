import React, { useState } from 'react'
import { connect } from 'react-redux'
import { NavigationScreenProp, NavigationState } from 'react-navigation'
import { AppError, reportError } from '../../../lib/errors'
import { ThunkDispatch } from '../../../store'
import { routeList } from '../../../routeList'
import { navigationActions } from '../../../actions'
import { Emoji, EmojiSection } from '../components/emojiSection'
import { Container } from '../../structure'
import { styles } from '../styles'
import { ScrollView } from 'react-native'
import { ChooseIssueSection } from '../components/chooseIssueSection'
import { DescriptionSection } from '../components/descriptionSection'
import { ContactSection } from '../components/contactSection'
import { GradientButton } from '../../structure/gradientButton'
import I18n from '../../../locales/i18n'
import strings from '../../../locales/strings'
interface PaymentNavigationParams {
  error: AppError | Error | undefined
}

interface Props extends ReturnType<typeof mapDispatchToProps> {
  navigation: NavigationScreenProp<NavigationState, PaymentNavigationParams>
}

export enum Inputs {
  None,
  Dropdown,
  Description,
  Contact,
}

const ErrorReportingContainer = (props: Props) => {
  const { navigateToScreen, navigation } = props

  const [pickedIssue, setIssue] = useState<string>()
  const [description, setDescription] = useState<string>('')
  const [contact, setContact] = useState<string>('')
  const [currentInput, setInput] = useState<Inputs>(Inputs.None)
  const [toggleState, setToggle] = useState<boolean>(false)
  const [selectedEmoji, setEmoji] = useState<Emoji>(Emoji.Empty)

  const onSubmitReport = () => {
    const { error } = props.navigation.state.params

    const userReport = {
      userError: pickedIssue,
      userDescription: description,
      userContact: contact,
    }
    console.log(userReport)

    if (navigation && error) {
      reportError({ ...userReport, error })
      if (error instanceof AppError) {
        navigateToScreen(error.navigateTo)
      } else {
        navigateToScreen(routeList.AppInit)
      }
    }
  }

  return (
    <Container style={styles.wrapper}>
      <ScrollView>
        <ChooseIssueSection
          currentInput={currentInput}
          pickedIssue={pickedIssue}
          onIssuePick={setIssue}
          setInput={setInput}
        />
        <DescriptionSection
          currentInput={currentInput}
          setInput={setInput}
          setDescription={setDescription}
          toggleState={toggleState}
          setToggle={setToggle}
          description={description}
        />
        <ContactSection
          onContactInput={setContact}
          currentInput={currentInput}
          setInput={setInput}
          contactValue={contact}
        />
        <EmojiSection selectedEmoji={selectedEmoji} setEmoji={setEmoji} />
        <GradientButton
          onPress={onSubmitReport}
          containerStyle={{
            marginTop: 45,
            marginBottom: 66,
          }}
          text={I18n.t(strings.SUBMIT_REPORT)}
        />
      </ScrollView>
    </Container>
  )
}

const mapDispatchToProps = (dispatch: ThunkDispatch) => ({
  navigateToScreen: (screen: routeList) =>
    dispatch(navigationActions.navigate({ routeName: screen })),
})

export const ErrorReporting = connect(
  null,
  mapDispatchToProps,
)(ErrorReportingContainer)
