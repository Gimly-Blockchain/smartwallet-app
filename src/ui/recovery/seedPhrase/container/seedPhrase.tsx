import React from 'react'
import { connect } from 'react-redux'
import { SeedPhrase as SeedPhraseComponent } from 'src/ui/recovery/seedPhrase/components/seedPhrase'
import { ThunkDispatch } from '../../../../store'
import { NavigationScreenProps } from 'react-navigation'
import { navigationActions } from '../../../../actions'
import { routeList } from '../../../../routeList'
import { StatusBar } from 'react-native'

interface Props
  extends ReturnType<typeof mapDispatchToProps>,
    ReturnType<typeof mapStateToProps>,
    NavigationScreenProps {}

export class SeedPhraseContainer extends React.Component<Props> {
  public render() {
    const mnemonic =
      this.props.navigation && this.props.navigation.getParam('mnemonic')
    return (
      <React.Fragment>
        <StatusBar
          animated={false}
          barStyle="light-content"
          showHideTransition="fade"
        />
        <SeedPhraseComponent
          seedPhrase={mnemonic}
          handleButtonTap={() => this.props.repeatSeedPhrase(mnemonic)}
        />
      </React.Fragment>
    )
  }
}

const mapStateToProps = () => ({})
const mapDispatchToProps = (dispatch: ThunkDispatch) => ({
  repeatSeedPhrase: (mnemonic: string) =>
    dispatch(
      navigationActions.navigate({
        routeName: routeList.RepeatSeedPhrase,
        params: { mnemonic },
      }),
    ),
})

export const SeedPhrase = connect(
  mapStateToProps,
  mapDispatchToProps,
)(SeedPhraseContainer)
