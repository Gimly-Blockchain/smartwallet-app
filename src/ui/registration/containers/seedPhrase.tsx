import React from 'react'
import { connect } from 'react-redux'
import { SeedPhrase as SeedPhraseComponent } from 'src/ui/registration/components/seedPhrase'
import { navigatorResetHome } from 'src/actions/navigation'
import { ThunkDispatch } from '../../../store'

interface Props
  extends ReturnType<typeof mapDispatchToProps>,
    ReturnType<typeof mapStateToProps> {
  navigation: { state: { params: any } } // TODO Type?
}

interface State {
  checked: boolean
}

export class SeedPhraseContainer extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props)
    this.state = {
      checked: false,
    }
  }

  render() {
    const { mnemonic } = this.props.navigation.state.params
    return (
      <SeedPhraseComponent
        seedPhrase={mnemonic}
        checked={this.state.checked}
        handleButtonTap={this.props.finishRegistration}
      />
    )
  }
}

const mapStateToProps = () => ({})
const mapDispatchToProps = (dispatch: ThunkDispatch) => ({
  finishRegistration: () => dispatch(navigatorResetHome()),
})

export const SeedPhrase = connect(
  mapStateToProps,
  mapDispatchToProps,
)(SeedPhraseContainer)
