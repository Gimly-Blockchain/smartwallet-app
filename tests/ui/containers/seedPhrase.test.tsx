import React from 'react'
import { SeedPhraseContainer } from 'src/ui/registration/containers/seedPhrase'
import { shallow } from 'enzyme'

describe('seedPhrase container', () => {
  it('mounts correctly and matches snapshot', () => {

    const finishRegistration = jest.fn()

    const props : SeedPhraseContainer['props'] = {
      finishRegistration,
      navigation: {
        state: {
          params: {
            mnemonic: 'mock seedPhrase',
          },
        },
      },
    }

    const rendered = shallow(<SeedPhraseContainer {...props} />)
    expect(rendered).toMatchSnapshot()
  })
})
