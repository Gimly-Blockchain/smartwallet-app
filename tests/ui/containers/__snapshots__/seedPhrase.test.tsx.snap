// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`seedPhrase container mounts correctly and matches snapshot 1`] = `
<Fragment>
  <StatusBar
    animated={false}
    barStyle="light-content"
    showHideTransition="fade"
  />
  <SeedPhrase
    handleButtonTap={[Function]}
    seedPhrase="mock seedPhrase"
  />
</Fragment>
`;
